### ISSUE TITLE
Parameter "body" is null on NavServer

### ISSUE THREAD

**Issue by luiscesjr** [godot-jolt/godot-jolt#659]
Hi, sometimes I get this error on my Ais around the game:

```
E 0:00:23:0244   Ai_Pathfinder_Kine.gd:247 @ _on_navigation_agent_3d_velocity_computed(): Parameter "body" is null.
  <C++ Source>   src\servers\jolt_physics_server_3d.cpp:1114 @ _body_get_direct_state()
  <Stack Trace>  Ai_Pathfinder_Kine.gd:247 @ _on_navigation_agent_3d_velocity_computed()
```

Since it's base directs me to Jolt server, I figured it could be it.

The code in question is:

```
func _on_navigation_agent_3d_velocity_computed(safe_velocity):
	velocity = velocity.move_toward(safe_velocity, .25)
	move_and_slide() #Line 247
```

And what body? While it does not crash the game, it floods the console with this error, with a couple dozens of them being thrown around.

And the code in the source is:

```
PhysicsDirectBodyState3D* JoltPhysicsServer3D::_body_get_direct_state(const RID& p_body) {
	JoltBodyImpl3D* body = body_owner.get_or_null(p_body);

	// Unlike most other server methods this one is meant to quietly return null if the body has
	// since been freed, which is used in places like `move_and_slide` to determine whether a
	// previously used platform has been freed or not.
	QUIET_FAIL_NULL_D(body);

	return body->get_direct_state();
}
```

Sometimes language is a barrier lol, but my Ai characters have not been freed for sure, which makes me wonder why it returns null?
Or is this supposed to happen, and can I suppress this without any worries?

**Comment by mihe** [comment]
I can't speak to whether you should be getting this error or not without seeing some kind of repro first, but the code you pasted from `JoltPhysicsServer3D` has not been released yet.

The equivalent method in version 0.9.0 emits an actual error when `body` is null, which is the error you're getting for some reason, whereas the code you pasted from the `master` branch does not actually emit any error (hence the `QUIET_*` macro). It was changed fairly recently as a result of #645.

I'll try and have a new release out this weekend, which will contain this change. But again, I don't know if that will hide an actual error in your case or just a false positive.

**Comment by luiscesjr** [comment]
Oh ok, I got it from the master branch here, thought was the same I was using.

Maybe it's the interactions from the bodies in my game then, I have some rigidbodies which queue_free after some set time limit.
But I don't see a way to check for that around their move_and_slide(), since there are no bodies to check there.

I get the collisions here:

```
func detectar_col_ocorrida() -> Vector3:
	for col_idx in get_slide_collision_count():
		var col := get_slide_collision(col_idx)
		if col.get_collider() is CharacterBody3D: 
			return col.get_normal().normalized()
		if col.get_collider() is StaticBody3D and col.get_normal().y < 0.1:
			return col.get_normal().normalized()
		if col.get_collider() is RigidBody3D:
			var force_multiplier = clamp(velocity.length(), 1.5, 10)
			var _force = -col.get_normal() * force_multiplier
			col.get_collider().apply_impulse(_force)
	return Vector3.ZERO
```

And, if any happened, I add them to:

```
	if col_vector != Vector3.ZERO:
		dir += col_vector * fator_peso_colisão # I probably need to normalize this later =D, you know, flying bodies...
	nav_agent.set_velocity(dir) #Which will then be calculated and used in the safe_velocity variable
```

I wonder, it's probably between this col detection and the safe velocity computed that I get this error if the body collided has been freed?

**Comment by mihe** [comment]
Assuming this issue is still affecting you, would you be able to try out the newly released [0.10.0-rc1](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.10.0-rc1) and see if that fixes the issue for you? If nothing else it should at least silence the error itself.

**Comment by mihe** [comment]
What I hope is the fix for this is part of the newly released [0.10.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.10.0-stable), so I'll go ahead and close this for now.

If it seems like the issue isn't resolved even when using 0.10.0-stable then feel free to bump this issue again.
